File: app/src/main/java/com/saienko/androidthings/barman/db/gpio/GpioDao.java
package com.saienko.androidthings.barman.db.gpio;

import android.arch.persistence.room.*;

import java.util.List;

/**
 * Created
 * User: Vasiliy Saienko
 * Date: 9/9/17
 * Time: 14:08
 */

@Dao
public interface GpioDao {
    @Query("SELECT * FROM gpio")
    List<Gpio> getAll();

    @Insert
    void insertAll(Gpio... gpios);

    @Insert
    void insert(Gpio gpio);

    @Delete
    void delete(Gpio gpio);

    @Query("DELETE FROM gpio")
    void clear();

    @Update
    void update(Gpio gpio);

    @Query("SELECT * FROM gpio WHERE gpioId NOT IN (:gpioIds)")
    List<Gpio> getAllFree(long[] gpioIds);

    @Query("SELECT * FROM gpio WHERE gpioId = :gpioId")
    Gpio get(long gpioId);
}

File: app/src/main/java/com/saienko/androidthings/barman/db/dao/GpioDao.kt
package com.saienko.androidthings.barman.db.dao

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Query
import com.saienko.androidthings.barman.db.entity.Gpio
import io.reactivex.Single

/**
 * Created
 * User: Vasiliy Saienko
 * Date: 9/9/17
 * Time: 14:08
 */

@Dao
interface GpioDao : BaseDao<Gpio> {

    @get:Query("SELECT * FROM gpio")
    val all: MutableList<Gpio>

    @Query("DELETE FROM gpio WHERE id = :gpioId")
    fun delete(gpioId: Long)

    @Query("DELETE FROM gpio")
    fun clear()

    @Query("SELECT * FROM gpio WHERE id NOT IN (:gpioIds)")
    fun getAllFree(gpioIds: LongArray): Single<MutableList<Gpio>>

    @Query("SELECT * FROM gpio WHERE id = :id")
    fun get(id: Long): Gpio
}
